### ISSUE TITLE
Detecting version of Swift

### ISSUE THREAD

**Issue by norio-nomura** [realm/SwiftLint#703]
I think following behaviors are usable:

``` console
$ TOOLCHAINS=com.apple.dt.toolchain.Swift_2_3 sourcekitten syntax --text "#sourceLocation()"
[
  {
    "offset" : 1,
    "length" : 14,
    "type" : "source.lang.swift.syntaxtype.identifier"
  }
]
$ TOOLCHAINS=com.apple.dt.toolchain.XcodeDefault sourcekitten syntax --text "#sourceLocation()"
[
  {
    "offset" : 0,
    "length" : 15,
    "type" : "source.lang.swift.syntaxtype.keyword"
  }
```

**Comment by norio-nomura** [comment]
We can disable rules that are needless on Swift 3 such as #701

**Comment by norio-nomura** [comment]
Are there any rules that we should disable on Swift 3 other than `conditional_binding_cascade` ?

**Comment by norio-nomura** [comment]
related to #700

**Comment by AmarJayR** [comment]
`type_name` should be disabled for Swift 3. Per Apple's Swift 3 API Design guidelines, enums should use lowerCamelCase. At the moment this triggers a `type_name` error.

**Comment by AliSoftware** [comment]
The rule telling the enum cases to be TitleCamelCased (instead of lowerCamelCased) is probably concerned too (`type_name` iirc). Wouldn't completely disable it though, just amend it to not consider `case foo` to be concerned by `type_name` as a `case` is considered an instance property, not a type.

**Comment by marcelofabri** [comment]
This was implemented on https://github.com/realm/SwiftLint/pull/1081. Thanks @norio-nomura for finding these behaviors!
